src/core/exception.d

```diff
@@ -563,25 +532,22 @@ extern (C) void onInvalidMemoryOperationError(void* pretend_sideffect = null) @t
 
 
 /**
- * A callback for switch errors in D.  A $(LREF SwitchError) will be thrown.
+ * A callback for switch errors in D.
  *
  * Params:
  *  file = The name of the file that signaled this error.
  *  line = The line number on which this error occurred.
- *
- * Throws:
- *  $(LREF SwitchError).
  */
 extern (C) void onSwitchError( string file = __FILE__, size_t line = __LINE__ ) @safe pure nothrow
 {
-    throw new SwitchError( file, line, null );
+    assert(0, "No appropriate switch clause found");
```

Comment:
One of the things I'd really like to do here, however, is make use of the `__FILE__` and `__LINE__` parameters in the error message.  That was the only reason I hadn't fixed this PR yet.  Given that such a trivial thing is actually quite difficult given the limitations discussed in https://github.com/dlang/druntime/pull/2268, however, I'd be willing to merge this PR as is, as long as there is a bugzilla issue filed to improve the error message.